### ISSUE TITLE
Exception if lambda function already exists

### ISSUE THREAD

**Issue by jxltom** [aws/chalice#731]
If a fuction already exists in AWS lambda, following error will be raised when running ```chalice deploy```

```
ERROR - While sending your chalice handler code to Lambda to create function
"somefunction", received the following error:

 An error occurred (ResourceConflictException) when calling the CreateFunction
 operation: Function already exist: somefunction
```

Will it be better if an option is added to enable overiding existed lambda functions? This is because it is very common to have existed lambda function due to deployment failure or deleting ```deployed.json``` accidentally.

**Comment by jxltom** [comment]
Or we can just deleting deployed lambda functions when ```chalice deploy``` fails due to any reasons such as incorrect setup of permissions, network issues and etc..

**Comment by dstufft** [comment]
Marking this as a feature request, thanks!

**Comment by romancini** [comment]
@jxltom , it isn't the best solution, but I versioned my .chalice/deployed/dev.json
Only with this file I could deploy my project in my aws enviroment with another computer.

**Comment by jxltom** [comment]
Yeah that's ok. But the dev.json may be deleted accidentally before committed to git. So with the proposed option, it saves a lot of trouble.
